frontend/src/libs/components/note/note.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  changeCursorPosition,
+  debounce,
+  getCursorPosition,
+  getValidClassNames,
+  sanitizeInput,
+} from '#libs/helpers/helpers.js';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useFormController,
+  useParams,
+  useRef,
+} from '#libs/hooks/hooks.js';
+import { type UserAuthResponseDto } from '#packages/users/users.js';
+import { actions as journalActions } from '#slices/journal/journal.js';
+
+import { NOTE_TIMEOUT } from './libs/constants.js';
+import { type NoteContent } from './libs/types.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  className: string;
+};
+
+const Note: React.FC<Properties> = ({ className }) => {
+  const { userId, selectedJournalEntry } = useAppSelector(
+    ({ auth, journal }) => {
+      return {
+        userId: (auth.authenticatedUser as UserAuthResponseDto).id,
+        selectedJournalEntry: journal.selectedJournalEntry,
+      };
+    },
+  );
+
+  const { isDirty, control } = useAppForm({
+    defaultValues: {
+      title: selectedJournalEntry?.title,
+      text: selectedJournalEntry?.text,
```

Comment:
we need to render this component only when selectedJournalEntry is present
```suggestion
      title: selectedJournalEntry.title,
      text: selectedJournalEntry.text,
```